pkg/project/project.go

```diff
@@ -45,21 +46,44 @@ func ReadProjectDescriptor(pathToFile string) (types.Descriptor, error) {
 
 	version := versionDescriptor.Project.Version
 	if version == "" {
+		fmt.Println("No schema version declared in project.toml, defaulting to schema version 0.1")
```

Comment:
The spec states that the file must contain a schema version:
https://github.com/buildpacks/pack/issues/1922#issuecomment-1737632210

As such, perhaps we should make this an error instead of a warning?

Reply:
@edmorley not if you go back far enough in the history (like [here](https://github.com/buildpacks/spec/blob/981e9f8b2daa2f6b219f0431edc03881dc61a78d/extensions/project-descriptor.md)). it seems like the spec only started mandating a schema version starting with the `0.2` project descriptor.  

if we allow for no schema version for backwards compatibility with the `0.1` format then you end up with this PR. i also have a version that enforces the schema version for the `0.1` format too but those changes would not be backwards compatible. 

@hone i noticed that the first appearance of the "MUST contain a schema version" text was in [this commit](https://github.com/buildpacks/spec/commit/9f1e939cfed0c64175bdcf782600201d417d7ac0) from you.  can you clarify if this restriction should apply only to the `0.2` format or both `0.2` and `0.1`?

Reply:
I agreed with @colincasey, I think one strategy could be:
1. We keep for now defaulting to 0.1 schema if no schema is present
2. At some point, we will deprecate 0.1 and maybe this message could be updated to inform the user they MUST declare the schema version, because support for 0.1 will be remove on pack X.Y.Z version
3. Once we deprecate support for 0.1 in pack X.Y.Z then we can fail the build